src/Modals/Modal.tsx

```diff
@@ -134,28 +142,49 @@ const Modal: FunctionComponent<ModalProps> = ({
   children,
   closeHandler,
   ariaLabelledBy,
+  forwardRef,
 }): ReactElement => {
   const theme = useContext(ThemeContext);
 
+  // If the ref is not provided, create one since it is used below
+  const finalForwardRef = forwardRef != null ? forwardRef : useRef(null);
+
   /**
    * Watch the isVisible props, and call the onOpen handler when it becomes
    * true. We're returning the onClose handler so it gets called during the
    * cleanup phase
    */
   useEffect(() => {
-    // restore everything to normal and call the onClose handler, if defined
-    const cleanup = (): void => {
-      document.body.style.overflow = '';
-      if (onClose !== undefined) {
-        onClose();
-      }
-    };
     if (isVisible) {
       // prevents the background from scrolling
       document.body.style.overflow = 'hidden';
       if (onOpen !== undefined) {
         onOpen();
       }
+      // Redirects the focus to the first focusable element in the modal when
+      // modal is visible
+      const listener = (event): void => {
+        const modal: HTMLElement = finalForwardRef.current;
+        if (!modal.contains(event.target as Node)) {
+          const focusables = 'button, [href], input, select, textarea,'
+            + ' [tabindex]:not([tabindex="-1"])';
```

Comment:
the focusable elements are buried inside the code so adding more elements would be a bit difficult.
It will be better if the list of the elements defined before the useEffect.

Reply:
That makes sense. I moved the list of focusable elements out of the `useEffect` in commit 2914c42.